tests/jointcalTestBase.py

```diff
@@ -0,0 +1,311 @@
+
+        old_rms_relative = rms_per_star(old_relative)
+        old_rms_absolute = rms_per_star(old_absolute)
+        new_rms_relative = rms_per_star(new_relative)
+        new_rms_absolute = rms_per_star(new_absolute)
+        print(len(dataRefs))
+        print("relative RMS (old, new):", old_rel_total*arcsec_per_radian,
+              new_rel_total*arcsec_per_radian)
+        print("absolute RMS (old, new):", old_abs_total*arcsec_per_radian,
+              new_abs_total*arcsec_per_radian)
+        plot_rms_histogram(plt, old_rms_relative, old_rms_absolute, new_rms_relative, new_rms_absolute,
+                           old_rel_total, old_abs_total, new_rel_total, new_abs_total)
+
+        # So one can muck-about with things after plotting...
+        import ipdb
+        ipdb.set_trace()
+
+
+def rms_per_star(data):
+    """
+    Compute the RMS/catalog star
+
+    @param data (dict) dict of starID:list(measurement deltas for that star)
+
+    @return (np.array) the rms for each starID (no guaranteed order)
+    """
+    result = np.empty(len(data))
+    for i, k in enumerate(data):
+        result[i] = np.sqrt(np.mean(np.array(data[k])**2))
+    return result
+
+
+def rms_total(data):
+    """
+    Compute the total rms across all stars.
+
+    @param data (dict) dict of starID:list(measurement deltas for that star)
+
+    @return (int) the total rms across all stars
+    """
+    n = 0
+    total = 0
+    for i, k in enumerate(data):
+        total += sum(np.array(data[k])**2)
+        n += len(data[k])
+    return np.sqrt(total/n)
+
+
+def plot_all_wcs_deltas(plt, dataRefs, visitCatalogs, old_wcss):
+    """Various plots of the difference between old and new Wcs."""
+
+    plot_all_wcs_quivers(plt, dataRefs, visitCatalogs, old_wcss)
+
+    # NOTE: uncomment this block to get per-ccd plots of the "distortion map".
+    # for i, ref in enumerate(dataRefs):
+    #     md = ref.get('calexp_md')
+    #     plot_wcs(plt, old_wcss[i], ref.get('wcs').getWcs(),
+    #              dim=(md.get('NAXIS1'), md.get('NAXIS1')),
+    #              center=(md.get('CRVAL1'), md.get('CRVAL2')), name='dataRef %d'%i)
+
+
+def wcs_convert(xv, yv, wcs):
+    """Convert two arrays of x/y points into an on-sky grid."""
+    xout = np.zeros((xv.shape[0], yv.shape[0]))
+    yout = np.zeros((xv.shape[0], yv.shape[0]))
+    for i, x in enumerate(xv):
+        for j, y in enumerate(yv):
+            sky = wcs.pixelToSky(x, y).toFk5()
+            xout[i, j] = sky.getRa()
+            yout[i, j] = sky.getDec()
+    return xout, yout
+
+
+def plot_all_wcs_quivers(plt, dataRefs, visitCatalogs, old_wcss):
+    """Make quiver plots of the WCS deltas for each CCD in each visit."""
+
+    for cat in visitCatalogs:
+        fig = plt.figure()
+        fig.set_tight_layout(True)
+        ax = fig.add_subplot(111)
+        for old_wcs, ref in zip(old_wcss, dataRefs):
+            if ref.dataId['visit'] == cat:
```

Comment:
Putting a single `if` statement in the `for` loop is something you should try to avoid (it's inefficient and it unnecessarily increases the indent level which makes it prone to errors).  Two options here: a lookup table mapping visit number to index in the `old_wcss` and `dataRefs` lists, or do `if ref.dataId['visit'] != cat: continue`.